sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubOptions.cs

```diff
@@ -0,0 +1,407 @@
+                }
+                _batchCheckpointFrequency = value;
+            }
+        }
+
+        private int _maxBatchSize;
+
+        /// <summary>
+        /// Gets or sets the maximum number of events delivered in a batch. Default 10.
+        /// </summary>
+        public int MaxBatchSize
+        {
+            get
+            {
+                return _maxBatchSize;
+            }
+
+            set
+            {
+                if (value < 1)
+                {
+                    throw new ArgumentException("Batch checkpoint frequency must be larger than 0.");
+                }
+                _maxBatchSize = value;
+            }
+        }
+
+        public bool InvokeProcessorAfterReceiveTimeout { get; set; }
+
+        public EventProcessorOptions EventProcessorOptions { get; }
+
+        private Action<ExceptionReceivedEventArgs> _exceptionHandler;
+
+        internal void SetExceptionHandler(Action<ExceptionReceivedEventArgs> exceptionHandler)
+        {
+            if (exceptionHandler == null)
+            {
+                throw new ArgumentNullException(nameof(exceptionHandler));
+            }
+
+            _exceptionHandler = exceptionHandler;
+        }
+
+        /// <summary>
+        /// Add an existing client for sending messages to an event hub.  Infer the eventHub name from client.path
+        /// </summary>
+        /// <param name="client"></param>
+        public void AddEventHubProducerClient(EventHubProducerClient client)
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+            string eventHubName = client.EventHubName;
+            AddEventHubProducerClient(eventHubName, client);
+        }
+
+        /// <summary>
+        /// Add an existing client for sending messages to an event hub.  Infer the eventHub name from client.path
+        /// </summary>
+        /// <param name="eventHubName">name of the event hub</param>
+        /// <param name="client"></param>
+        public void AddEventHubProducerClient(string eventHubName, EventHubProducerClient client)
+        {
+            if (eventHubName == null)
+            {
+                throw new ArgumentNullException(nameof(eventHubName));
+            }
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            _clients[eventHubName] = client;
+        }
+
+        /// <summary>
+        /// Add a connection for sending messages to an event hub. Connect via the connection string.
+        /// </summary>
+        /// <param name="eventHubName">name of the event hub. </param>
+        /// <param name="sendConnectionString">connection string for sending messages. If this includes an EntityPath, it takes precedence over the eventHubName parameter.</param>
+        public void AddSender(string eventHubName, string sendConnectionString)
```

Comment:
We probably want to move to `Producer` to align with the new terminology.    This was a big ask from the service team during T2 API reviews.